kubetest2/internal/deployers/eksapi/k8s.go

```diff
@@ -93,15 +94,25 @@ func isNodeReady(node *corev1.Node) bool {
 	return false
 }
 
-const awsAuthMapRolesPrefix = `
-- username: system:node:{{EC2PrivateDNSName}}
+func getAwsAuthMapRolesPrefix(osDistro string) string {
+	var nodeNameFlavor string
+	if strings.EqualFold(osDistro, "al2023") {
+		nodeNameFlavor = "SessionName"
+	} else {
+		nodeNameFlavor = "EC2PrivateDNSName"
+	}
+	return fmt.Sprintf(`
+- username: system:node:{{%s}}
   groups:
     - system:bootstrappers
     - system:nodes
-  rolearn: `
+  rolearn: `, nodeNameFlavor)
+  
+}
```

Comment:
I think we should use a proper go template for this, since we're filling in 2 values now. Examples in the deployers/eksapi/templates package